File: Source/Fantasy.Jobs/TaskItemScalarReader.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Fantasy.Jobs.Properties;

namespace Fantasy.Jobs
{
    public class TaskItemScalarReader : ObjectWithSite, ITagValueProvider 
    {
        #region ITagValueProvider Members

        public char Prefix
        {
            get { return '%'; }
        }

        public string GetTagValue(string tag, IDictionary<string, object> context)
        {
            string[] expr = tag.Split(new char[] { '.' }, 2, StringSplitOptions.RemoveEmptyEntries);

            IJob job = (IJob)this.Site.GetService(typeof(IJob));

            TaskItem[] items = job.GetEvaluatedItemsByCatetory(expr[0]);

            IEnumerable<string> values;
            if (expr.Length > 1)
            {
                values = items.Select(x => x[expr[1]]);
            }
            else
            {
                values = items.Select(x => x.Name);
            }

            string rs = values.Where(x => !string.IsNullOrEmpty(x)).FirstOrDefault();
            return rs ?? string.Empty;

        }

        public bool HasTag(string tag, IDictionary<string, object> context)
        {
            return true;
        }

        public bool IsEnabled(IDictionary<string, object> context)
        {
            return (bool)context.GetValueOrDefault("EnableTaskItemReader", true) &&
                this.Site != null && this.Site.GetService(typeof(IJob)) != null;
        }

        #endregion
    }
}

File: AutoConverted/Fantasy.Jobs/TaskItemScalarReader.java
package Fantasy.Jobs;

import Fantasy.Jobs.Properties.*;

public class TaskItemScalarReader extends ObjectWithSite implements ITagValueProvider
{
//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#region ITagValueProvider Members

	public final char getPrefix()
	{
		return '%';
	}

	public final String GetTagValue(String tag, java.util.Map<String, Object> context)
	{
		String[] expr = tag.split(new char[] { '.' }, 2, StringSplitOptions.RemoveEmptyEntries);

		IJob job = (IJob)this.Site.GetService(IJob.class);

		TaskItem[] items = job.GetEvaluatedItemsByCatetory(expr[0]);

		Iterable<String> values;
		if (expr.length > 1)
		{
//C# TO JAVA CONVERTER TODO TASK: Lambda expressions and anonymous methods are not converted by C# to Java Converter:
			values = items.Select(x => x[expr[1]]);
		}
		else
		{
//C# TO JAVA CONVERTER TODO TASK: Lambda expressions and anonymous methods are not converted by C# to Java Converter:
			values = items.Select(x => x.getName());
		}

//C# TO JAVA CONVERTER TODO TASK: There is no Java equivalent to LINQ queries:
//C# TO JAVA CONVERTER TODO TASK: Lambda expressions and anonymous methods are not converted by C# to Java Converter:
		String rs = values.Where(x => !DotNetToJavaStringHelper.isNullOrEmpty(x)).FirstOrDefault();
		return (rs != null) ? rs : "";

	}

	public final boolean HasTag(String tag, java.util.Map<String, Object> context)
	{
		return true;
	}

	public final boolean IsEnabled(java.util.Map<String, Object> context)
	{
		return (boolean)context.GetValueOrDefault("EnableTaskItemReader", true) && this.Site != null && this.Site.GetService(IJob.class) != null;
	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#endregion
}